onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1923,6 +1923,248 @@ Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-
         FusedMatMulShapeInference(ctx);
       });
 
+    static const char* NonZero_doc = R"DOC(
+  Behaves similar to https://github.com/onnx/onnx/blob/master/docs/Operators.md#NonZero
+  returns a tensor that contains coordinates for non-zero entries. Even though the input
+  is a sparse tensor, that usually does not contain zeros, the model logic may construct
+  sparse tensor containing explicit zeros that convey model specific meaning.
+  If the input sparse tensor has 2-D COO indices, the output shape would be (nnz,2) with 2 coordinates per
+  non-zero entry. If the input sparse tensor contains 1-D COO indices, the output would contain flat indices
+  of shape (nnz).
+
+  XXX: Should be make the output shape fixed regardless on the input COO indices shape? 
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(NonZero)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "2-dimensional sparse matrix A. Either COO format either 1-D or 2-D index", "T")
+      .Output(0, "B", "1 or 2-dimensional dense matrix B", "T1")
+      .TypeConstraint("T", {"sparse_tensor(float)"}, "Constrain to float at this time")
+      .TypeConstraint("T1", {"tensor(int64)"}, "Index type")
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        // Output type is index type
+        propagateElemTypeFromDtypeToOutput(ctx, TensorProto_DataType_INT64, 0);
+        // Output shape is runtime dependent and can not be inferred as it depends on
+        // the input sparse tensor COO indices shape.
+      }) ;
+
+    static const char* SparseGemm_doc = R"DOC(
+  Behaves as to https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gemm
+  XXX: can we do without transpose now?
+  XXX: It has a lot of common code with SparseToDenseMatMul which we can reuse.
```

Comment:
We are fine with either Gemm or MatMul. I missed your comment earlier on the word doc about this.

There are two use cases for Gemm Sparse support:
1. SubsetByIndices
Given a Sparse Tensor input, return a Sparse subset of the Tensor by a set of input indices.
Our plan is to use Gather or MatMul.

2. FilterNonZeroValues
Given a Sparse Tensor input, return a Sparse Tensor without zero values.
Our plan is to use Gemm/MatMul + Gather.